pkg/controller/servicebindingrequest/binder.go

```diff
@@ -303,31 +304,67 @@ func (b *Binder) appendEnvVar(
 
 // appendEnvFrom based on secret name and list of EnvFromSource instances, making sure secret is
 // part of the list or appended.
-func (b *Binder) appendEnvFrom(envList []corev1.EnvFromSource, secret string) []corev1.EnvFromSource {
+func (b *Binder) appendEnvFrom(envList []corev1.EnvFromSource, binddata string) []corev1.EnvFromSource {
+
+	if b.sbr.Spec.BindingReferenceType == nil {
+		return envList
+	}
+	dataType := b.sbr.Spec.BindingReferenceType.Kind
+
 	for _, env := range envList {
-		if env.SecretRef.Name == secret {
-			b.logger.Debug("Directive 'envFrom' is already present!")
-			// secret name is already referenced
-			return envList
+		if dataType == ConfigMapKind {
+			if env.ConfigMapRef.Name == binddata {
+				b.logger.Debug("Directive 'envFrom' is already present!")
```

Comment:
We should log whether envFrom here is for `ConfigMap` or `secret`

Reply:
Done, thank you!